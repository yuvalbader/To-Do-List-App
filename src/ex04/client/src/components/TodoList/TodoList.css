.search_bar {
  margin-top: 15px;
}
