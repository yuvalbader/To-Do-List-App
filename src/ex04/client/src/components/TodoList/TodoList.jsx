import React from "react";
import ListItem from "../ListItem/ListItem";
import "../TodoList/TodoList.css";
import PropTypes from "prop-types";

export default function TodoList({
  items,
  handleDoneTask,
  handleDelete,
  handleEdit,
}) {
  return (
    <div className="todo-list-container">
      <ul className="todo-list">
        {items.map((item, index) => (
          <ListItem
            key={index}
            isDone={item.isDone}
            item={item}
            handleDoneTask={handleDoneTask}
            handleDelete={handleDelete}
            handleEdit={handleEdit}
          />
        ))}
      </ul>
      <label className="pending-tasks">
        {items.length === 0
          ? "You don't have pending tasks"
          : `You have ${items.length} tasks`}
      </label>
    </div>
  );
}

TodoList.prototype = {
  items: PropTypes.array.isRequired,
  handleDoneTask: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
};
