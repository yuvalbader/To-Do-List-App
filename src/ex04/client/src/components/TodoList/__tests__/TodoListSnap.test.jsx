import renderer from "react-test-renderer";
import TodoList from "../TodoList";
import { store } from "../../../Redux/store";


test("Render correctly empty tasks", () => {
  const tree = renderer
    .create(<TodoList store={store}  />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
