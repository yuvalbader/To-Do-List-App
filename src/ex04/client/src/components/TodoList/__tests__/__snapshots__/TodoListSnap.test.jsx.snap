// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly empty tasks 1`] = `
<div
  className="todo-list-container"
>
  <div
    aria-busy={false}
    className="input-component monday-storybook-search_size search_component_wrapper"
    role="search"
  >
    <div
      className="input-component__label--wrapper"
    >
      <div
        className="input-component__input-wrapper input-component__input-wrapper--size-medium"
      >
        <input
          aria-activedescendant=""
          aria-invalid={null}
          aria-label="Search your todo"
          aria-owns=""
          autoComplete="off"
          className="search_bar search_component input-component__input input-component__input--has-icon"
          data-testid="search_search"
          disabled={false}
          id="search"
          maxLength={null}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search your todo"
          readOnly={false}
          required={false}
          role=""
          type="search"
          value=""
        />
        <div
          className="monday-style-clickable input-component__icon--container input-component__icon--container-has-icon input-component__icon--container-active monday-style-clickable--disable-text-selection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex="-1"
        >
          <svg
            aria-hidden={true}
            aria-label="Search"
            className="icon_component input-component__icon icon_component--no-focus-style"
            fill="currentColor"
            height="18px"
            onClick={[Function]}
            viewBox="0 0 20 20"
            width="18px"
          >
            <path
              clipRule="evenodd"
              d="M8.65191 2.37299C6.9706 2.37299 5.35814 3.04089 4.16927 4.22976C2.9804 5.41863 2.3125 7.03108 2.3125 8.7124C2.3125 10.3937 2.9804 12.0062 4.16927 13.195C5.35814 14.3839 6.9706 15.0518 8.65191 15.0518C10.0813 15.0518 11.4609 14.5691 12.5728 13.6939L16.4086 17.5303C16.7014 17.8232 17.1763 17.8232 17.4692 17.5303C17.7621 17.2375 17.7622 16.7626 17.4693 16.4697L13.6334 12.6333C14.5086 11.5214 14.9913 10.1418 14.9913 8.7124C14.9913 7.03108 14.3234 5.41863 13.1346 4.22976C11.9457 3.04089 10.3332 2.37299 8.65191 2.37299ZM12.091 12.1172C12.9878 11.2113 13.4913 9.98783 13.4913 8.7124C13.4913 7.42891 12.9815 6.19798 12.0739 5.29042C11.1663 4.38285 9.9354 3.87299 8.65191 3.87299C7.36842 3.87299 6.1375 4.38285 5.22993 5.29042C4.32237 6.19798 3.8125 7.42891 3.8125 8.7124C3.8125 9.99589 4.32237 11.2268 5.22993 12.1344C6.1375 13.0419 7.36842 13.5518 8.65191 13.5518C9.92736 13.5518 11.1509 13.0483 12.0568 12.1514C12.0623 12.1455 12.0679 12.1397 12.0737 12.134C12.0794 12.1283 12.0851 12.1227 12.091 12.1172Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="monday-style-clickable input-component__icon--container input-component__icon--container-has-icon monday-style-clickable--disable-text-selection"
          data-testid="clean-search-button_search"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex="-1"
        >
          <svg
            aria-hidden={true}
            aria-label="Clear Search"
            className="icon_component input-component__icon icon_component--no-focus-style"
            fill="currentColor"
            height="18px"
            onClick={[Function]}
            viewBox="0 0 20 20"
            width="18px"
          >
            <path
              clipRule="evenodd"
              d="M6.53033 5.46967C6.23744 5.17678 5.76256 5.17678 5.46967 5.46967C5.17678 5.76256 5.17678 6.23744 5.46967 6.53033L8.62562 9.68628L5.47045 12.8415C5.17756 13.1343 5.17756 13.6092 5.47045 13.9021C5.76334 14.195 6.23822 14.195 6.53111 13.9021L9.68628 10.7469L12.8415 13.9021C13.1343 14.195 13.6092 14.195 13.9021 13.9021C14.195 13.6092 14.195 13.1343 13.9021 12.8415L10.7469 9.68628L13.9029 6.53033C14.1958 6.23744 14.1958 5.76256 13.9029 5.46967C13.61 5.17678 13.1351 5.17678 12.8422 5.46967L9.68628 8.62562L6.53033 5.46967Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <ul
    className="todo-list"
  />
  <label
    className="pending-tasks"
  >
    You don't have pending tasks
  </label>
</div>
`;
